### DISCUSSION TITLE
Modify the request header in index.php

### DISCUSSION THREAD

**Discussion by fd6130** [symfony/symfony#43099] [Q&A] [answered]
Before using runtime component, i can do something like this for proxy header:

```
$kernel = new Kernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG']);
$request = Request::createFromGlobals();

if ($request->server->has('HTTP_X_ORIGINAL_HOST')) // for trusted proxy
{
    $request->server->set('HTTP_X_FORWARDED_HOST', $request->server->get('HTTP_X_ORIGINAL_HOST'));
    $request->headers->set('X_FORWARDED_HOST', $request->server->get('HTTP_X_ORIGINAL_HOST'));
}

$response = $kernel->handle($request);
$response->send();
$kernel->terminate($request, $response);
```

But after runtime component, it simplified to this:

```
use App\Kernel;

require_once dirname(__DIR__).'/vendor/autoload_runtime.php';

return function (array $context) {
    return new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']);
};
```

How can i modify the request header in index.php using runtime component?

**Comment by michaoj** [comment, marked as answer]
I see two options to achive that : 

1. Create a listener on kernel.request to update the current response 
2. Use the $_SERVER superglobal on you r index.php (but I don't  like this option)

**Reply by fd6130** [reply]
Thanks. Option 1 are more reliable. And need to set higher priority for that listener otherwise the request won't change.
